### PR TITLE
Use hiopVector for regularizations

### PR DESCRIPTION
**Pull request by nychiang**
Change regularizations `delta_wx`, `delta_wd`,`delta_cc` and `delta_cd` from scalars to vectors

### CODE REVIEW

**Comment by nychiang** [comment]
@cnpetra please have a look of `hiopKKTLinSysSparseCondensed.cpp`, especially the part where you add `delta_cc` to `delta_wx`.

File: src/Utils/hiopOptions.cpp [hunk 1]
```diff
@@ -984,9 +984,18 @@ void hiopOptionsNLP::register_options()
...
 
-    vector<string> range(2); range[0]="unified"; range[1]="randomized";
-    register_str_option("dual_reg_method",
-                        "unified",
+    vector<string> range = {"primal_first", "dual_first"};    
+    register_str_option("normaleqn_regularization_priority",
+                        "dual_first",
+                        range,
+                        "When normal equation is used and the iterate matrix is not p.d., updating dual regularization "
+                        "is more efficient thatn updateing the primal ones. Setting this option to `primal_first` will "
```
Comment (cnpetra) [inline src/Utils/hiopOptions.cpp:992]: "than updating"

**Review by cnpetra — APPROVED** [review]
LGTM, just left a couple of small comments

File: src/Optimization/hiopKKTLinSysSparseNormalEqn.cpp [hunk 2]
```diff
@@ -259,17 +259,15 @@ bool hiopKKTLinSysSparseNormalEqn::build_kkt_matrix(const double& delta_wx_in,
...
-  assert(delta_cc>=0.0);
+  
   Diag_reg_->setToZero();
-  if(delta_cc > 0.0) {
-    if(nlp_->options->GetString("dual_reg_method") == "unified") {
-      dual_reg_->setToConstant(delta_cc);
-    } else if(nlp_->options->GetString("dual_reg_method") == "randomized") {
-      dual_reg_->set_to_random_uniform(0.9*delta_cc, 1.1*delta_cc);
-    }
+  if(!delta_cc_in.is_zero() || !delta_cd_in.is_zero()) {
```
[the remark this answers is not in the document]
Comment (nychiang) [reply src/Optimization/hiopKKTLinSysSparseNormalEqn.cpp:264]: this is fixed in #520, see https://github.com/LLNL/hiop/pull/520/files

**Comment by nychiang** [comment]
@cnpetra all your comments have been addressed.

**Pull request merged**